_drafts/2017-07-05-what-is-a-headless-cms.md

```diff
@@ -0,0 +1,178 @@
+---
+title: What is a headless cms anyway?
+author: borsi
+image: "/img/posts/headless-cover.jpg"
+tags: [enterprise content management, headless cms, api, restful]
+---
+
+The term 'headless cms' has been rising in popularity for a while now. But why the hype and what is it all about?
+
+---
+## What is a headless CMS?
+First, it's worth mentioning how traditional "monolithic" cms's work. Wordpress or Drupal are usually included in this category. This means that the modal layer is tightly coupled with the view, and the creators of the platform provide you with a templating engine with which you can implement your frontend. So in this model you'd have:
+
+- a Database at the "bottom", where you store your content
+- an App, with a UI that lets you [Create, Read, Update and Delete][25358749] this content
+- and a View layer which displays the content (usually by using templates)
+
+  [25358749]: https://en.wikipedia.org/wiki/Create,_read,_update_and_delete "CRUD"
+
+
+ Older versions of [sensenet ECM][ed1adc41] are also following this philosophy, although the platform is more geared towards enterprise features, such as [content versioning][a19fba28] and [collaboration][206457f6].
+ Thus, the 'E' in ECM is for Enterprise.
+
+  [ed1adc41]: https://www.sensenet.com/product "Sensenet ECM Product features"
+  [a19fba28]: http://wiki.sensenet.com/Versioning_and_approval "Versioning and approval"
+  [206457f6]: http://wiki.sensenet.com/Workspace "Workspaces in sensenet ECM"
+
+This is why you could do something like this in [sensenet 6.x][9f950a61], or using [sn-webpages 7.x][46ac6b1f]:
+
+```xml
+<div class="sn-article-list">
+    <%foreach (var content in this.Model.Items)
+      { %>
+    <div class="sn-article-list-item">
+	<h2 class="sn-article-title">
+		<a href="<%=Actions.BrowseUrl(content)%>">
+			<%= content.DisplayName %>
+		</a>
+	</h2>
+        <small class="sn-article-info">
+		PublishedBy: <%= content["Author"] %> on 
+                <span class='<%= "sn-date-" + content["Id"] %>'>
+			<%= content["ModificationDate"]%>
+		</span>
+	</small>
+        <div class="sn-article-lead">
+            <%=content["Lead"] %>
+        </div>
+    </div>
+    <%} %>
+</div>
+```
+<p align="center">
+*fig 1. ~ The template is tightly coupled with the data it needs to present.*
+</p>
+
+  [46ac6b1f]: https://github.com/SenseNet/sn-webpages "Sn-webpages"
+  [9f950a61]: http://community.sensenet.com/docs/how-to-install-sn6/ "Install sensenet 6.5"
+
+In a headless CMS (or ECMS for that matter), though, the backend does not care about how the content is displayed. It just returns with the data needed for the frontend through a RESTful API.
+
```

Comment:
A headless CMS is clearly a content-focused datasource, from which you can build content-based applications.